### DISCUSSION TITLE
Medusa and offline media folders - support question

### DISCUSSION THREAD

**Discussion by johnvick** [pymedusa/Medusa#9378] [Q&A]
Branch: masterCommit: d0c136d7a528a471b51676140bd35d24d97f65c6Version: 0.5.10Database: 44.18
--
```
Python Version: | 3.7.6 (default, Dec 21 2019, 02:28:04) [GCC 8.3.0]
SSL Version: | OpenSSL 1.1.1d 10 Sep 2019
OS: | Linux-5.8.0-45-generic-x86_64-with
Locale: | en_US.UTF-8
User: | root
Program Folder: | /app/medusa
Config File: | /config/config.ini
Database File: | /config/main.db
Cache Folder: | /config/cache
Log Folder: | /config/Logs
Arguments: | --nolaunch --datadir /config
Web Root: | /medusa
Runs in Docker: | Yes
```

Asked this in reddit bot suggested post here.

I have Medusa Docker install on small Intel device running Ubuntu 20.04. The downloads land on this device and are then post-processed to a separate OMV system into NFS shares - this is only powered up in the evenings.

What happens is that Medusa only snatches new episodes when the OMV device is powered up. I'm sure it used to snatch 24/7 but at some point this changed and I'm unsure why.

Any way to have Medusa snatch 24/7?

**Comment by medariox** [comment]
Medusa has no knowledge whether the OMV device is powered or not. If the NFS share is available the file will be moved, otherwise it will fail. My guess would be that your OMV changed it's sale state. That's probably where you want to check to find the issue.

**Reply by johnvick** [reply]
Thanks for your input. It's not a case of the file being moved to OMV which obviously won't happen if OMV if off. It's that the download does not start at all, and this step does not involve OMV unless Medusa checks to see if the final resting place for  the file, an OMV nfs share, is available or not and then decides to proceed or not. What's meant by the term "sale state"?

**Reply by p0psicles** [reply]
Can you enable debug logs, then manual search and snatch an episode, while your Nas is not available. 

AFAIK it should snatch, but postprocessing could be an issue if your Nas is offline.

**Comment by johnvick** [comment]
When the NAS is offline Medusa runs (verified by Docker ps) and the web interface comes up  - but shows are not displayed so I can't manually search and snatch. After a while I get "cannot connect to Medusa" message from the browser.  With the NAS on it all comes back to normal. It seems like Medusa is not fully functional when the NAS and therefore the nfs shares containing the episodes are offline.

**Reply by johnvick** [reply]
Just looked at the log in Docker/Portainer with NAS off. This appeared:
2021-03-25 06:50:28 INFO     SEARCHQUEUE-BACKLOG-253138 :: [d0c136d] Downloading ***.***.S02E02.1080p.HDTV.x264.HUNTED with 2 seeders and 0 leechers and size 1.22 GB from 1337
Nothing happens with Deluge until I power up NAS then it starts to download.

**Reply by medariox** [reply]
But that has nothing to do with Medusa anymore at that point? Maybe Deluge checks if the NFS share is available before starting the download?

**Comment by johnvick** [comment]
Deluge is on a remote Seedbox and has no way of checking the NFS shares.

**Reply by johnvick** [reply]
Key key problem seem to be that Medusa is not sending the link to the remote Deluge or Deluge isn't receiving it. I've changed to the black hole method as will see how that goes.

**Reply by johnvick** [reply]
Same behaviour with blackhole method. Today added a new show, confirmed it had added properly, then powered off NAS. Checked Medusa logs to confirm it was running on the other device. Left it two hours, nil snatched. Powered up NAS and within 5 minutes it had snatched 10 episodes.  Searching two weeks of logs for the term SNATCHQUEUE-SNATCH I am pretty sure it only occurs when the NAS is powered up with the phrase most commonly appearing just after 4 pm when the NAS powers up automatically.

**Comment by johnvick** [comment]
I set up a new Ubuntu server 20.10 mini PC and a new Docker install of Medusa. It connects to the same OMV NAS that contains the media files in NFS shares. Exactly the same happens - Medusa does not snatch when the NAS is off. Power it up and it all happens. The other thing that doesn't work properly when the NAS is off is the web interface, and this is the same on both mini PCs. Having duplicated the problem on two devices I wonder could there be be a bug in Medusa causing this i.e. it won't snatch when the final destination for the download isn't there? Or have I got something misconfigured?

**Reply by p0psicles** [reply]
Do you have a full debug log of the snatch? Is the torrent send?

**Comment by johnvick** [comment]
Last night enabled debug log, added a new show, checked it had added OK, powered off the NAS.

Go up today nothing in Deluge. Powered up NAS about 7:10 AM and soon after the snatching started.
[medusa-log.txt](https://github.com/pymedusa/Medusa/files/6216158/medusa-log.txt)

**Reply by p0psicles** [reply]
it's not a debug log. Prob needed to restart medusa.
But I see this warning.. `2021-03-28 07:14:19 WARNING  SNATCHQUEUE-SNATCH-368166 :: [] Deluge: Unable to send Torrent`.
So there is probably going something wrong with deluge, not accepting the torrent. You could try the black hole method in stead. That way you 100% sure now medusa is not the limitation.

**Reply by johnvick** [reply]
I'll try again with proper debug log. Deluge works 100% when NAS powered up (it's on a remote seedbox). I tried blackhole earlier - same story, only worked when NAS powered up.

**Reply by p0psicles** [reply]
Let me put it this way. When you use black hole, your essentialli downloading the torrent to a folder. I asume the black hole folder is not on the Nas but the application server. 

Does medusa Store the torrent file into the folder?

**Reply by johnvick** [reply]
Thanks for you help with this BTW.

The blackhole folder is on the same device as Medusa and is sync'd to the blackhole folder on the seedbox where Deluge picks it up. Works 100% when NAS powered up.

[Medusa log2.txt](https://github.com/pymedusa/Medusa/files/6216422/Medusa.log2.txt)

Tried again - new show added and confirmed at around 8:15 am. NAS powered off. Out for bike ride, came home powered up NAS just before 10 AM. Downloads and Telegram notifications started.

**Reply by johnvick** [reply]
I have replaced the remote Deluge with Transmission installed on the same device as Medusa. Exactly the same happens which eliminates Deluge as a possible culprit. NAS powered up at 12:30 pm and then the downloads start.
[Medusa-log3.txt](https://github.com/pymedusa/Medusa/files/6216609/Medusa-log3.txt)

**Reply by p0psicles** [reply]
Does the torrent get into the black hole folder?

**Reply by johnvick** [reply]
Yes but only after the NAS is powered up. The torrent files have date stamp a minute after NAS power up time.

I'm sure this is new behaviour, Medusa used to snatch 24/7 a few months back. I also used to be able to access the web interface with the NAS off (I still can but with reduced functionality)  . I can't be the only one to have a setup with Medusa on a low power device or Pi on 24/7 and the media files on a NAS only powered up when used? Anyone else had similar problems?

**Reply by p0psicles** [reply]
Please replicate. Turn Nas off. Then a snatch (manually) an episode. And check what happens to the torrent in the black hole folder.

**Reply by johnvick** [reply]
I can't access the Medusa web interface with the NAS off so can't do this sorry.  I've been setting up a new show, checking it has loaded OK in Medusa then powering off NAS. No snatches happen until the NAS is back on.

**Reply by p0psicles** [reply]
So your 100÷ sure that your nas only is used for the show folders. Not for medusa prog folder, and not for data folder. So all medusa related is on a local -not mapped- drive?

**Reply by p0psicles** [reply]
Can you contact me on discord?

**Comment by johnvick** [comment]
100%. And teste on two Intel mini-PCs same result.

**Reply by johnvick** [reply]
100% sure. I’m in NZ Monday morning at work how about this evening NZ time? I’ll be away from home but can access the devices remotely.

**Reply by p0psicles** [reply]
Let's continue on discord

**Reply by johnvick** [reply]
OK tomorrow evening (Tue NZ time) would be good I'll be back home so much easier than using motel wifi. I have Discord account same username.

**Reply by p0psicles** [reply]
Just leave a message in the general channel when you have time

**Comment by johnvick** [comment]
No luck on Discord - maybe just a final word here and then I'll leave it unless there are further ideas.

I have tested this problem with three Medusa client devices and two NAS with NFS shares holding the media files. I've installed Medusa with Docker and git-pull. I've mounted the NFS shares with fstab and autofs. I've used local Transmission, blackhole and remote seedbox deluge for downloading. 

Same problems with any combination of above - Medusa doesn't snatch, the web interface doesn't work and restarting Medusa via docker does not work if NFS shares are not available. Start the NAS, then everything works. 

The only conclusion I can reach is Medusa does not operate fully if the target folders for the TV episodes are on another device that is not available. Debug logs are posted above. 

It's not a big deal I can power up the NAS on a timer each day and even if away from home the episodes keep coming in.

Thanks for the input so far.

**Comment by johnvick** [comment]
Reporting I have got to the bottom of this with with a pointer from p0ps. The problem is running Medusa in a Docker container with NFS shares only works when the shares are available - if not Medusa does not fully function. I believe this is a general problem and not specific to Medusa. Whether the shares are mounted on the host system with fstab, autofs or in the container using the native Docker mounting system makes no difference. My solution is to run a script on cron just after the NAS powers off:

#!/bin/sh
/bin/systemctl stop autofs 
sleep 5
/usr/bin/docker restart medusa
/bin/systemctl start autofs 

And when the NAS is back on:

/usr/bin/docker restart medusa.

This gives Medusa full functionality when shares are not available.

Hope this helps someone else as it bugged me for weeks.
